### PR TITLE
GitHub Actions to add/remove label "stale" and comment

### PR DESCRIPTION
**Pull request by totoprayogo1916**
**Description**
GA concept:
- Any changes made in the main branch will be automatically reviewed against all open pull requests (PRs). If a conflict arises, a label "stale" will be added, and a comment will be left to notify the contributors.
- Upon an update to a PR, a reevaluation will be conducted. If the PR has the "stale" label and there are no conflicts, the label will be removed.

Live Test: https://github.com/totoprayogo1916/github-playground/pull/13

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: .github/workflows/label-add-conflict.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Auto Add Label "stale" & Comment Conflicts
+
+on:
+  push:
+    branches:
+      - develop
+      - '4.*'
```
Comment (ddevsr) [inline .github/workflows/label-add-conflict.yml:7]: No needed

```suggestion
```

**Review by kenjis — APPROVED** [review]
Oh, thank you! This seems good.

File: .github/workflows/label-add-conflict.yml (hunk 1 above)
Comment (ddevsr) [reply .github/workflows/label-add-conflict.yml:7]: Oh okay. Seems smooth

**Review by ddevsr — APPROVED** [review]

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
This workflow was labeled for PRs with no conflicts, so I reverted.
https://github.com/codeigniter4/CodeIgniter4/pull/7947#issuecomment-1726974403
https://github.com/codeigniter4/CodeIgniter4/pull/7944#issuecomment-1726974445
https://github.com/codeigniter4/CodeIgniter4/pull/7925#issuecomment-1726974479

File: .github/workflows/label-add-conflict.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          IFS=$'\n' # Set Internal Field Separator to newline to handle array elements
+
+          # Iterate through the PRs in PR_LIST
+          for pr in $(echo "$PR_LIST" | jq -c '.[]'); do
+            mergeable=$(echo "$pr" | jq -r '.mergeable')
+            author=$(echo "$pr" | jq -r '.author.login')
+            labels=$(echo "$pr" | jq -c '.labels[].name' | tr -d '[]"')
+            url=$(echo "$pr" | jq -r '.url')
+
+            if [ "$mergeable" != "MERGEABLE" ] && [[ ! "$labels" == *"stale"* ]]; then
```
[the remark this answers is not in the document]
Comment (kenjis) [reply .github/workflows/label-add-conflict.yml:46]: But if it is "UNKNOWN", we cannot put a label.

At least, I can see the outcomes:
```
$ gh pr list -L 100 --json mergeable,url 
[
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7951"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7947"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7946"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7944"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7933"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7925"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7924"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7846"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7732"
  },
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7724"
  },
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7718"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7717"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7491"
  },
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/7404"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/6877"
  },
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/6845"
  },
  {
    "mergeable": "CONFLICTING",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/6284"
  },
  {
    "mergeable": "MERGEABLE",
    "url": "https://github.com/codeigniter4/CodeIgniter4/pull/1797"
  }
]
```

[the remark this answers is not in the document]
Comment (kenjis) [reply .github/workflows/label-add-conflict.yml:46]: Yes, it is safe.

[the remark this answers is not in the document]
Comment (kenjis) [reply .github/workflows/label-add-conflict.yml:46]: Thanks. Can you send another PR with `"$mergeable" == "CONFLICTING"`?

**Pull request merged**